### DISCUSSION TITLE
Segmentation fault when running samples/simple on RHEL9 using ubi docker image

### DISCUSSION THREAD

**Discussion by ottobackwards** [marcelotduarte/cx_Freeze#2483] [Q&A] [answered]
I have been unable to get the samples to run with 7.1.1 on a rhel9 ubi image. 
It errors with segmentation fault.

Has anyone gotten this to work?

python 3.9.18
gcc 11.4.1

**Comment by ottobackwards** [comment]
This repo has a Dockerfile and steps to reproduce

https://github.com/ottobackwards/cx_freeze_rhel9_fail

**Comment by marcelotduarte** [comment, marked as answer]
For me it is working:
```Dockerfile
FROM registry.access.redhat.com/ubi9/ubi:latest as builder

RUN dnf clean all && dnf -y update && dnf install -y git gcc python3-pip wget && \
  python3 -m pip install cx_freeze

WORKDIR /root
RUN wget https://github.com/marcelotduarte/cx_Freeze/archive/refs/tags/7.1.1.tar.gz && tar -xzvf 7.1.1.tar.gz

WORKDIR /root/cx_Freeze-7.1.1/samples/simple
RUN python3 setup.py build

FROM registry.access.redhat.com/ubi9/ubi:latest
COPY --from=builder /root/cx_Freeze-7.1.1/samples/simple/build/exe.linux-x86_64-3.9 /test/
ENTRYPOINT ["/test/hello"]
```
To run:
`podman build -t test2483 .`
`podman run --rm test2483:latest`

I used podman, but should work with docker.
```
Hello from cx_Freeze
The current date is June 26, 2024 16:28:07

Executable: /test/hello
Prefix: 
Default encoding: utf-8
File system encoding: ascii

ARGUMENTS:
/test/hello

PATH:
/test/lib/library.zip
/test/lib
```

**Reply by ottobackwards** [reply]
hrm.

I adapted your changes to the docker file, and ran both with docker and Podman, and still got the segfault.
I'm am running docker and Podman on apple silicon with `--platform linux/amd64`, I wonder if that is the difference?
Seems strange that that would be an issue though.
I'll see if I can find another platform to test on.

**Reply by marcelotduarte** [reply]
The error may be in the emulator since you are on Apple. I tested in Ubuntu 22.04.
Regarding the Dockerfile, it is recommended to separate the builder from the execution, so that there is no collision between the installed version and the version used by cx_Freeze, of some libraries, such as OpenSSL, and Tkinter.

**Comment by ottobackwards** [comment]
ok, I ran on a vm instead of docker and it worked.  glad that worked, but not sure what the issue could be.
thanks for your help
